src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityResourceIntTest.kt

```diff
@@ -949,62 +984,170 @@ class ActivityResourceIntTest : IntegrationTestBase() {
       }
 
       @Test
-      fun `should retain the end date if pay rate start date is moved back at same time as rate is changed`() {
+      fun `should not error when checking pay rates if an offender without an incentive level is allocated`() {
```

Comment:
This time I've added a few tests with the corrupted data that has been created in NOMIS while the sync was unaware of DPS pay rate start dates. It was this data that broke the previous attempted fix - but we should have handled such data anywhere and have ended up with a simpler solution.